PostSharp.Community.StructuralEquality.Weaver/OperatorInjection.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+using PostSharp.Sdk.CodeModel;
+using PostSharp.Sdk.CodeModel.Helpers;
+using PostSharp.Sdk.Extensibility;
+
+namespace PostSharp.Community.StructuralEquality.Weaver
+{
+    /// <summary>
+    /// Overwrites equality operators with a call to the equality method.
+    /// </summary>
+    public class OperatorInjection
+    {
+        private readonly IGenericMethodDefinition weaveMethod;
+        private readonly IGenericMethodDefinition staticEqualsMethod;
+
+        public OperatorInjection( Project project )
+        {
+            var objectTypeDef = project.Module.Cache.GetIntrinsic( IntrinsicType.Object ).GetTypeDefinition();
+            this.staticEqualsMethod = project.Module.FindMethod( objectTypeDef, "Equals", 2 );
+            
+            var operatorTypeDef = project.Module.Cache.GetType( typeof(Operator) ).GetTypeDefinition();
+            this.weaveMethod = project.Module.FindMethod( operatorTypeDef, "Weave" );
+        }
+        
+        private void ProcessOperator( TypeDefDeclaration enhancedType, string operatorName, string operatorSourceName, bool negate )
+        {
+            var operatorMethod = GetOperatorMethod( enhancedType, operatorName );
+
+            string operatorExample = $"public static bool operator {operatorSourceName}({enhancedType.Name} left, {enhancedType.Name} right) => Operator.Weave(left, right);";
+            
+            if ( operatorMethod == null )
+            {
+                throw new Exception($"The equality operator was not found on type {enhancedType.Name}, implement it like this: {operatorExample}");

```

Comment:
Here and elsewhere, instead of throwing an exception, we probably want to use Message.Write and somehow safely abort, because PostSharp, unlike Fody, reacts to SDK exceptions by pushing the stack trace into the Error window.